src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  };
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = !this.hidden;
+  };
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    const indexOfAnimal = Animal.alive.indexOf(animal);
+
+    if (animal.hidden === true || animal instanceof Carnivore) {
+      return;
+    }
```

Comment:
try to rewrite this condition to avoid unnecessary `return` - write code for `bite` case.